apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -560,11 +584,84 @@ public void registerSpanListener(ActivationListener activationListener) {
         return activationListeners;
     }
 
-    void registerLifecycleListeners(List<LifecycleListener> lifecycleListeners) {
+    synchronized void start(List<LifecycleListener> lifecycleListeners) {
         this.lifecycleListeners.addAll(lifecycleListeners);
         for (LifecycleListener lifecycleListener : lifecycleListeners) {
             lifecycleListener.start(this);
         }
+        if (activeConfigOptionSet) {
+            tracerState = TracerState.RUNNING;
+            logger.info("Tracer switched to RUNNING state");
+        } else {
+            pause();
+        }
+    }
+
+    public synchronized void stressDetected() {
```

Comment:
Naming, maybe `onStressDetected` would be more explicit, same for `onStressRelieved` below.